asserts/internal/grouping.go

```diff
@@ -169,13 +221,56 @@ func (gr *Groupings) Deserialize(label string) (*Grouping, error) {
 	return &g, nil
 }
 
+func (gr *Groupings) bitsetDeserialize(g *Grouping, b []byte) (*Grouping, error) {
+	buf := bytes.NewBuffer(b)
+	binary.Read(buf, binary.LittleEndian, &g.size)
+	if g.size > gr.maxGroup+1 {
+		return nil, errSerializedLabel
+	}
+	if g.size <= gr.bitsetThreshold {
+		// should not have used a bitset repr for so few elements
+		return nil, errSerializedLabel
```

Comment:
The two error cases here sound like internal errors? Should we have separate errors for them? Do we have enough context to know what is the failing grouping when this error bubbles up?

Reply:
To clarify, I mean enough context in the logs for us if we need to debug an error.

Reply:
they are not strictly snapd internal, these serializations are used as keys and sent to the store and back (the store should treat as opaque though), the most likely case for this error is probably using internal.Serialize directly in tests badly. I think a single errSerializedLabel is mostly for convenience, I suppose we could have distinct fmt.Errorf for each but seems follow up material.

Reply:
Ack, thank you.